File: Entities/Appointment.cs
namespace PrivateHospitalSystem.Entities
{
    public enum AppointmentStatus
    {
        Scheduled,
        Completed,
        Cancelled
    }

    public class Appointment
    {
        public Guid Id { get; set; }

        public Guid PatientId { get; set; }
        public Patient? Patient { get; set; }

        public Guid DoctorId { get; set; }
        public Doctor? Doctor { get; set; }

        public Guid RoomId { get; set; }
        public Room? Room { get; set; }

        public DateTime ScheduledAt { get; set; }
        public int DurationMinutes { get; set; } = 30; 

        public AppointmentStatus Status { get; set; } = AppointmentStatus.Scheduled;
        public string? Notes { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: frontend/src/types/appointment.ts
export interface Appointment {
  id: string
  patientId: string
  patientName: string
  doctorId: string
  doctorName: string
  roomId: string
  roomNumber: string
  scheduledAt: string
  durationMinutes: number
  status: string
  notes: string | null
}

export interface CreateAppointmentDto {
  patientId: string
  doctorId: string
  roomId: string
  scheduledAt: string
  durationMinutes: number
  notes?: string
}